rfcs/convergence/first-rule-of-aria.md

```diff
@@ -0,0 +1,204 @@
+    </TableRow>
+  </TableBody>
+</Table>
+
+
+// renders to
+
+<table>
+  <thead>
+    <tr>
+      <th colspan="2">The table header</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>The table body</td>
+      <td>with two columns</td>
+    </tr>
+  </tbody>
+</table>
+```
+
+Since table is a common example of virtualization scenarios, where elements, content, headers, etc,. can be added on the go by JS code, the structural behavior provided by semantic elements can be easily compromised. For example in the case where a div is added between `tbody` and `tr` for scrolling purposes.
+
+```tsx
+<Table>
+  <TableHead>
+    <TableRow>
+      <TableCell colSpan={2}>
+        The table header
+      </TableCell>
+    </TableRow>
+  </TableHead>
+  <TableBody>
+    <div>
+      <TableRow>
+        <TableCell>
+          The table body
+        </TableCell>
+        <TableCell>
+          with two columns
+        </TableCell>
+      </TableRow>
+    </div>
+  </TableBody>
+</Table>
+
+<table>
+  <thead>
+    <tr>
+      <th colspan="2">The table header</th>
+    </tr>
+  </thead>
+  <tbody>
+    <!-- This div breaks the markup, since tbody cannot contain  -->
+    <div>
+      <tr>
+        <td>The table body</td>
+        <td>with two columns</td>
+      </tr>
+    </div>
+  </tbody>
+</table>
+```
+
+The idea of going for semantic first approach is to ensure that in the default cases, where the user of the lib simply doesn't care about edge cases such as `a` element inside of a `button` element, semantic elements will be used ensuring a better overall experience across different environments, and in the edge cases there's an option to dispose the semantic element for something more generic and ARIA.
+
+The example above could easily be converted to ARIA:
+
+```tsx
+// This as="div" is informs all compound components to opt-out of semantic-first approach
+<Table as="div">
+  <TableHead>
+    <TableRow>
+      <TableCell colSpan={2}>
+        The table header
+      </TableCell>
+    </TableRow>
+  </TableHead>
+  <TableBody>
+    <div>
+      <TableRow>
+        <TableCell>
+          The table body
+        </TableCell>
+        <TableCell>
+          with two columns
+        </TableCell>
+      </TableRow>
+    </div>
+  </TableBody>
+</Table>
+
+<div role="table">
+  <div role="rowgroup">
+    <div role="row">
+      <span role="columnheader" aria-colspan="2">The table header</span>
+    </div >
+  </div>
+  <div role="rowgroup">
+    <div>
+      <div role="row">
+        <span role="cell">The table body</span>
+        <span role="cell">with two columns</span>
+      </div>
+    </div>
+  </div>
+</div>
+```
+
+This proposal does not introduce any API, is just a pattern to be followed when creating converged components, and valid to add that in extreme cases, like the table, that semantic first approach is simply not viable, than this should be ignored.
+
+### Pros and Cons
+
+#### Pros
+
+- Better support overall for A11y
```

Comment:
well since we support `as` I can't see a way around not reeimplementing them smwhere in the codebase :/

Reply:
I'd say that's not the "topic" of this PR? 👀

Reply:
I understand your point of keeping the PR focused, we don't have to discuss it here. I am afraid though that if in this PR we come to a decision to support native elements, there might be a follow up RFC/discussion on how to handle these types of default behaviours when dealing with composition. The most common problem that affects almost every design framework is 'I want a div button because it's easier to style'

Reply:
Definetely we'll be needing some discussion on opting out of semantic elements alternatives, such as `as` property, and how to ensure native behaviour.
After all, even if we make a button a `button` by default the user might transform it into a `div`, and in that case we got to provide handlers to ensure A11y.